### ISSUE TITLE
get Windows compatibility

### ISSUE THREAD

**Issue by ltboye** [STAMP-project/dspot#651]
#### Characteristics
- **Issue Type**: bug
- **Reproducibility**: always
- **Severity**: major
- **Tool/Service/Component**: DSpot 1.2.2-SNAPSHOT (escape-dot-only-if-they-exist)
- **Execution Environment**: Windows 10
- **Reporter**: Lars Thomas

#### Description
Running on Windows, I was unable to specify multiple amplifiers.

--amplifiers MethodAdd:MethodRemove as in documentation seems to be interpreted as a a single amplifier.

@danglotb suggested --amplifiers MethodAdd;MethodRemove. It ran fine for some minutes, but then ended with error:

MethodRemove : The term 'MethodRemove' is not recognized as the name of a cmdlet, function, script file, or operable program. Check the spel
ling of the name, or if a path was included, verify that the path is correct and try again.
At line:1 char:138
+ ... ta.model.ModelTest --timeOut 1000 --amplifiers MethodAdd;MethodRemove
+                                                              ~~~~~~~~~~~~
    + CategoryInfo          : ObjectNotFound: (MethodRemove:String) [], CommandNotFoundException
    + FullyQualifiedErrorId : CommandNotFoundException

See https://github.com/STAMP-project/dspot-usecases-output/tree/master/Tellu/TelluLib-ModelTest-Descartes-Amplifiers2

**Comment by danglotb** [comment]
Hi @ltboye 

It looks like the `;` is not recognized as an argument separator by as a command separator.

I think Windows is trying to run the command `MethodRemove`, and obviously, this command does not exist.

I do not understand why this is happenning.

I do not feel that this issue is related to DSpot, but rather to the command line.

**Comment by monperrus** [comment]
@ltboye looking forward to your progress 

Plenary meeting, for the record: if we cannot get it working on Windows in two weeks, we would run it in a VM instead

**Comment by ltboye** [comment]
I seem to have tracked down the most serious issue with Windows compatibility, that of DSpot not running at all, the error being that Maven does not find the Java compiler. I had this error on two of my three machines, and now I have introduced it on the last, which so far had an older version of Java 8.

I updated the Java sdk from 1.8.0_121 to the latest, 1.8.0_202.
Previously reported by mvn -version, where DSpot would run:
Java version: 1.8.0_121, vendor: Oracle Corporation, runtime: C:\Program Files\Java\jdk1.8.0_121\jre
After update:
Java version: 1.8.0_202, vendor: Oracle Corporation, runtime: C:\Program Files\Java\jdk1.8.0_202\jre
After this update, DSpot fails with the familiar error "No compiler is provided in this environment. Perhaps you are running on a JRE rather than a JDK?". Note that as always, Maven still works just fine for building, its just with DSpot we get this error. JAVA_HOME is "C:\Program Files\Java\jdk1.8.0_202". I don't know why Maven adds \jre to the path, but as said, it works outside of DSpot.

It seems that something has changed in the jdk at some point, after which point Maven no longer works through DSpot on Windows. I have no idea how to proceed with this. Searches for this common error just gives results related to JAVA_HOME, which is not the problem here. This needs someone with Maven expertise. I don't want to downgrade to some old version of the jdk.
